gc/base/MemorySubSpace.cpp

```diff
@@ -1929,3 +1936,28 @@ MM_MemorySubSpace::wasContractedThisGC(uintptr_t gcCount)
 	return (gcCount == _extensions->heap->getResizeStats()->getLastHeapContractionGCCount());
 }
 
+#if defined(OMR_GC_IDLE_HEAP_MANAGER)
+uintptr_t
+MM_MemorySubSpace::releaseFreeMemoryPages(MM_EnvironmentBase* env)
+{
+	Assert_MM_unreachable();
+        return 0;
+}
+
+bool
+MM_MemorySubSpace::isSystemGCAllowed(MM_EnvironmentBase* env, uint32_t gcCode)
+{
+	bool invokeGC = true;
+	if (J9MMCONSTANT_IMPLICIT_GC_IDLE_GC == gcCode) {
+		MM_MemorySpace* defaultMemorySpace = _extensions->heap->getDefaultMemorySpace();
+		uintptr_t freeMemorySize = defaultMemorySpace->getApproximateActiveFreeMemorySize(MEMORY_TYPE_OLD);
+		uintptr_t actvMemorySize = defaultMemorySpace->getActiveMemorySize(MEMORY_TYPE_OLD);
+		uintptr_t previousMemorySize = 0 < _extensions->lastGCFreeBytes ? _extensions->lastGCFreeBytes : actvMemorySize;
+
+		if ((0 < freeMemorySize) && (20 > (((previousMemorySize - freeMemorySize) * 100) / actvMemorySize))) {
+			invokeGC = false;
+		}
```

Comment:
I a previous commented on the criteria, without realizing that that invokeGC is being set to false (for some reason I assumed true). Now the logic makes more sense, so  you can disregard my previous comment in the original form (which have been updated meanwhile)